doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+
+	Control the capability of user by changing the role
+* **change password**
+
+	Change the password of a user.
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='ArchitectureDesign'></a>Architecture Design
+![Arch_diagram](Arch_diagram.png)
+
+(flow description in the chapter below)
+
+###  2.2 <a name='High-LevelDesign'></a>High-Level Design
+
+Sonic has two main users:
+* Admin: Administrator with limited access rights.It is part of sudo group.<br/>
+* Root: Superuser with highest access rights. Login is disabled for this user.
+
+We will define two roles for users:<br/>
+* **admin**: a role for administrator users with write and read permissions.<br/>
+	Each admin user will be part of the following groups:<br/>
+	**primary group**: admin<br/>
+	**secondary groups**: sudo,docker,redis,adm
+
+* **monitor**: a role for monitoring users with read permissions only. <br/>
+	Each monitor user will be part of the following groups:<br/>
+	**primary group**: adm<br/>
+	**secondary groups**: N/A
+
+In addition, we will define two default users:
+* **admin**: User role is admin.<br/>
+	The only change from the current admin in Sonic is the additional group: adm
+
+* **monitor**: User role is monitor. <br/>
+	User with monitoring permissions only .<br/>
+
+**Note**:
+ Group adm is used for system monitoring tasks. Members of this group can read the files under "/var/log"
+
+
+###  2.3 <a name='UsercfdRole'></a>Usercfgd Role:
+
+Usercfgd is the daemon that get events from USER_TABLE in CONF_DB and apply the new configuration on the system.<br/>
+The new daemon should start running after "config-setup.service" (i.e. after database is ready) but before systemd enables "system-logind.service" (i.e. before user can login to system)<br/>
+
+
+![services_diagram](services_diagram.png))<br/>
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+```
+USER_TABLE:{
+  "<username>":{
+    "state": {{enabled/disabled}}
+    "full-name": {{string}}
+    "hashed-password": {{string}}
+    "hashed-password_history": {{string}}
+    "role": {{admin/monitor}}
+    "passwd_date_of_last_change": {{uint32}}
+  }
+}
+```
+
+```
+ROLE_TABLE:{
+  "<role-name>":{
+    "primary_group": {{string}}
+    "secondary_groups": {{string}}
+  }
+}
+```
+###  3.2 <a name='ConfigDBschemas'></a>ConfigDB schemas
+
+```
+; Defines schema for User configuration attributes in USER_TABLE:
+key                                   = "<username>"                ;user configuration
+; field                               = value
+STATE                                 = "enabled" / "disabled"      ; user enable/disable
+FULL-NAME                             = STRING                      ; Full-name/Description of user
+HASHED-PASSWORD                       = STRING                      ; Hashed password
+HASHED-PASSWORD_HISTORY               = STRING                      ; List of old passwords (0-99)
```

Comment:
Do we need a "expires" parameter?

Reply:
See the comment above: [https://github.com/sonic-net/SONiC/pull/1048/files#r1073802100](https://github.com/sonic-net/SONiC/pull/1048/files#r1073802100)